### PR TITLE
[PHEE-564] Webclient and Callback Adapter Added

### PR DESCRIPTION
**Pull request by abhinavm117**
## Description

* Describe the changes made and why they were made.
* Add a link to teh design document or include the design bullet points related to this PR here.

 _(Ignore if these details are present on the associated JIRA ticket)_

## Checklist

Please make sure these boxes are checked before submitting your pull request - thanks!

- [x] Design related bullet points or design document link related to this PR added in the description above.

- [x] Updated corresponding Postman Collection or Api documentation for the changes in this PR.

- [x] Create/update unit or integration tests for verifying the changes made.

- [x] Add required Swagger annotation and update API documentation with details of any API changes if applicable

- [x] Followed the naming conventions as given in https://docs.google.com/document/d/1Q4vaMSzrTxxh9TS0RILuNkSkYCxotuYk1Xe0CMIkkCU/edit?usp=sharing

### CODE REVIEW

File: src/main/java/org/mifos/connector/common/webclient/WebClientAdapter.java [hunk 1]
```diff
@@ -0,0 +1,128 @@
...
+
+    // GET
+    public <T> CompletableFuture<T> getRequest(String url, HttpHeaders headers, Class<T> responseType) {
+        log.info("Executing GET request to URL: {}", url);
+        return webClient.get()
+                .uri(url)
+                .headers(httpHeaders -> httpHeaders.addAll(headers))
+                .retrieve()
+                .bodyToMono(responseType)
+                .doOnSuccess(responseBody -> log.info("GET request to {} succeeded, response Body {}", url, responseBody))
```
Comment (fynmanoj) [inline src/main/java/org/mifos/connector/common/webclient/WebClientAdapter.java:32]: use debug

File: src/main/java/org/mifos/connector/common/webclient/WebClientAdapter.java [hunk 2]
```diff
@@ -0,0 +1,128 @@
...
+    public <T, R> CompletableFuture<R> postRequest(String url, T body, HttpHeaders headers, MediaType mediaType, Class<R> responseType) {
+        log.info("Executing POST request to URL: {}", url);
+        return webClient.post()
+                .uri(url)
+                .contentType(mediaType)
+                .headers(httpHeaders -> httpHeaders.addAll(headers))
+                .body(Mono.just(body), body.getClass())
+                .retrieve()
+                .bodyToMono(responseType)
+                .doOnSuccess(responseBody -> log.info("POST request to {} succeeded with response {}", url, responseBody))
```
Comment (fynmanoj) [inline src/main/java/org/mifos/connector/common/webclient/WebClientAdapter.java:51]: use debug

File: src/main/java/org/mifos/connector/common/webclient/WebClientAdapter.java [hunk 3]
```diff
@@ -0,0 +1,128 @@
...
+
+    public <R> CompletableFuture<R> postRequest(String url, HttpHeaders headers, MediaType mediaType, Class<R> responseType) {
+        log.info("Executing POST request to URL: {} without a body", url);
+        return webClient.post()
+                .uri(url)
+                .contentType(mediaType)
+                .headers(httpHeaders -> httpHeaders.addAll(headers))
+                .retrieve()
+                .bodyToMono(responseType)
+                .doOnSuccess(responseBody -> log.info("POST request to {} succeeded with response {}", url, responseBody))
```
Comment (fynmanoj) [inline src/main/java/org/mifos/connector/common/webclient/WebClientAdapter.java:68]: use dubug

**Review by fynmanoj — CHANGES_REQUESTED** [review]

File: src/main/java/org/mifos/connector/common/webclient/WebClientAdapter.java (hunk 1 above)
Comment (abhinavm117) [reply src/main/java/org/mifos/connector/common/webclient/WebClientAdapter.java]: done

File: src/main/java/org/mifos/connector/common/webclient/WebClientAdapter.java (hunk 2 above)
Comment (abhinavm117) [reply src/main/java/org/mifos/connector/common/webclient/WebClientAdapter.java]: done

File: src/main/java/org/mifos/connector/common/webclient/WebClientAdapter.java (hunk 3 above)
Comment (abhinavm117) [reply src/main/java/org/mifos/connector/common/webclient/WebClientAdapter.java]: done

Comment (abhinavm117) [reply src/main/java/org/mifos/connector/common/webclient/WebClientAdapter.java]: done

File: src/main/java/org/mifos/connector/common/webclient/WebClientAdapter.java (hunk 2 above)
Comment (abhinavm117) [reply src/main/java/org/mifos/connector/common/webclient/WebClientAdapter.java]: done

**Pull request merged**